src/nft/ZoraCreator1155Impl.sol

```diff
@@ -372,15 +372,35 @@ contract ZoraCreator1155Impl is
     /// @param tokenId The token ID to mint
     /// @param quantity The quantity of tokens to mint
     /// @param minterArguments The arguments to pass to the minter
-    function mint(IMinter1155 minter, uint256 tokenId, uint256 quantity, bytes calldata minterArguments) external payable nonReentrant {
+    /// @param finder The user or platform that finds the sale
+    /// @param lister The originator of the sale
+    function mint(
+        IMinter1155 minter,
+        uint256 tokenId,
+        uint256 quantity,
+        bytes calldata minterArguments,
+        address finder,
+        address lister
+    ) external payable nonReentrant {
         // Require admin from the minter to mint
         _requireAdminOrRole(address(minter), tokenId, PERMISSION_BIT_MINTER);
 
-        // Get value sent and handle mint fee
-        uint256 ethValueSent = _handleFeeAndGetValueSent(quantity);
+        uint256 remainingEth = _parseAttachedEth(msg.value, quantity);
+
+        // If this is a free mint:
+        if (remainingEth == 0) {
```

Comment:
seems like these are two distinct paths and new features, can you make sure a new test exists for each of them, to demonstrate the new features work properly and their edge cases properly tested?  It would also improve the documentation.